Dependencies/xr/Include/XR.h

```diff
@@ -78,6 +86,17 @@ namespace xr
         bool IsValid{true};
     };
 
+    using NativePlaneIdentifier = void*;
+    struct Plane
+    {
+        Pose Center{};
+        float* Polygon{};
```

Comment:
Broadly speaking, I think we should try to avoid raw pointers in general, but especially in contracts as much as possible. Would `std::vector<float>` work here?

Reply:
The value we get back from ARCore must be pushed into a dynamically allocated float array, so doing it in a std::vector requires us to both dynamically allocate the array then copy it over to a std::vector, but I can do that if we prefer.

Reply:
Worked this out offline, I'm now using an std::vector<float> and using std::vector::resize along with .data() to accomplish this.

Reply:
Yea I agree, in general it seems like there is too much manual memory management happening here (easy to make mistakes and have leaks). It seems like if we use `std::vector<Plane>`, `std::vector<float>`, etc. we can avoid a lot of the manual allocations/deallocations.